MdePkg/Library/StackCheckLib/Readme.md

```diff
@@ -15,44 +15,77 @@
 
 Because UEFI doesn't use the C runtime libraries provided by MSVC, the stack
 check code is written in assembly within this library. GCC and Clang compilers
 have built-in support for stack cookie checking, so this library only handles failures.
 
-### StackCheckLibStaticInit
+The stack cookie value is initialized at compile time via updates to the AutoGen process.
+Each module will define `STACK_COOKIE_VALUE` which is used for the module stack cookie
+value.
 
-`StackCheckLibStaticInit` is an instance of `StackCheckLib` which does not update the
-stack cookie value for the module at runtime. It's always preferable to use
-`StackCheckLibDynamicInit` for improved security but there are cases where the stack
-cookie global cannot be written to such as in execute-in-place (XIP) modules and during
-the Cache-as-RAM (CAR) phase of the boot process. The stack cookie value is initialized
-at compile time via updates to the AutoGen process. Each module will define
-`STACK_COOKIE_VALUE` which is used for the module stack cookie value.
+The entry point libraries under `MdePkg/DynamicStackCookieEntryPointLib/` update the stack
+cookie value at runtime for improved security, but there are cases where the stack cookie
+global cannot be written to such as in execute-in-place (XIP) modules and during the
+Cache-as-RAM (CAR) phase of the boot process. It is always preferable to use
+one of the dynamic stack cookie entry points when possible.
 
-### StackCheckLibDynamicInit
+### StackCheckLib
 
-This section is future work. The below is the proposed instance.
+`StackCheckLib` provides the stack cookie checking functionality per architecture and
+toolchain. The currently supported pairs are IA32{GCC,MSVC}, X64{GCC, MSVC},
+ARM{GCC, MSVC}, and AARCH64{GCC, MSVC}. `StackCheckLib` is agnostic as to
+whether the stack cookie was updated during build time or run time; it simply
+checks the cookie in the MSVC case and in both GCC and MSVC responds to stack
+cookie checking failures.
 
-`StackCheckLibDynamicInit` is an instance of `StackCheckLib` which updates the stack
-cookie value for the module at runtime. This is the preferred method for stack cookie
-initialization as it provides improved security. The stack cookie value is initialized
-at runtime by calling `GetRandomNumber32()` or `GetRandomNumber64()` to generate a random
-value via the platform's random number generator protocol. If the random number generator
-returns an error, then the value will still have the build-time randomized value to fall
-back on.
+To add support for other architectures/toolchains, additional assembly files
+should be added to `StackCheckLib.inf` and scoped to that architecture/toolchain.
+
+Note: SEC and PEI_CORE typically should not perform stack cookie checks because
+exceptions are not handled at this point, so any failure will result in a triple
+fault. Generally platforms should use `StackCheckLibNull` for these phases, except

```

Comment:
Triple fault is an IA32/X64 concept and should be removed.  Also, many CPU archs do have exception handlers in these early phases, they just do very simple actions like dump the CPU state because exceptions are not expected.  Platforms could implement exception handling in these early phases if needed.  I think it would be better to state that a stack cookie check failures generate exceptions, and that stack cookies should only be enabled if the platform provides exception handlers for stack cookie check failures.

Reply:
Sure, I will update this.